CesiumGltf/test/TestMetadataConversions.cpp

```diff
@@ -192,6 +252,18 @@ TEST_CASE("Test MetadataConversions for float") {
         std::string_view()));
   }
 
+  SECTION("returns std::nullopt for invalid strings") {
```

Comment:
Complexity comment. 

This section and the one preceding are almost the same, other than one converts from `string` and the other from `string_view`.

Can we restructure to reduce code, but crucially reduce the duplication of input test data?

Ex.
```
 REQUIRE(!MetadataConversions<float, std::string_view>::convert(std::string_view("10.00f hello")));
  ...
 REQUIRE(!MetadataConversions<float, std::string>::convert("10.00f hello"));
```
becomes
```
TestStringConversion<float>("10.00f hello", std::nullopt);

void TestStringConversion<toVal> (const std::string& input, std::optional<toVal> result) {
   REQUIRE(MetadataConversions<toVal, std::string>::convert(input) == result);
   REQUIRE(MetadataConversions<toVal, std::string_view>::convert(std::string_view(input)) == result);
}
```

This thought could extend to all tests. For example, it looks like all numerical conversions fail with mixed types of input, or input with invalid characters. It would be nice to have a defined set of "invalid conversion input" that we could reuse for all the related tests.

The ultimate goal: easier to maintain code with more test coverage